### PR TITLE
Remove redundant returns [#2477]

### PR DESCRIPTION
**Pull request by Kristiyan96**
Based on [[#2477]](https://github.com/ethereum/ethereum-org-website/issues/2477#issue-814586365)

- Remove redundant returns in map blocks
- Remove redundant returns in filter blocks
- Simplify render of stateless components
- Refactor if-else logic in two obvious cases

### CODE REVIEW

File: src/components/ButtonLink.js [hunk 1]
```diff
@@ -45,19 +45,15 @@ const Secondary = styled(StyledButton)`
   }
 `
```
Comment (Kristiyan96) [inline src/components/ButtonLink.js:47]: changed the if-else logic in this file

File: src/components/Eth2TableOfContents.js [hunk 2]
```diff
@@ -77,29 +77,16 @@ const TableOfContentsLink = ({ depth, item }) => {
   )
 }
```
Comment (Kristiyan96) [inline src/components/Eth2TableOfContents.js:79]: Simplified render logic, if-else if map made no sense

**Comment by samajammin** [comment]
Wow, thanks for this @Kristiyan96!

@wackerow please take a look when available.

**Review by wackerow — APPROVED** [review]
@Kristiyan96 LGTM! Fixed some conflicts that arose. Thanks for going through these!

@samajammin good to merge?

**Review by samajammin — APPROVED** [review]
Thanks for this! 🙌

**Comment by samajammin** [comment]
@all-contributors please add @Kristiyan96 for code.

**Pull request merged**